src/verilog/verilog_expr.h

```diff
@@ -1946,4 +1946,45 @@ to_verilog_implicit_typecast_expr(exprt &expr)
   return static_cast<verilog_implicit_typecast_exprt &>(expr);
 }
 
+class verilog_past_exprt : public binary_exprt
+{
+public:
+  verilog_past_exprt(exprt __what, exprt __ticks)
+    : binary_exprt(__what, ID_verilog_past, std::move(__ticks), __what.type())
+  {
+  }
+
+  const exprt &what() const
+  {
+    return op0();
+  }
+
+  exprt &what()
+  {
+    return op0();
+  }
+
+  const exprt &ticks() const
+  {
+    return op1();
+  }
+
+  exprt &ticks()
+  {
+    return op1();
+  }
+};
+
+inline const verilog_past_exprt &to_verilog_past_expr(const exprt &expr)
+{
+  verilog_past_exprt::check(expr);
+  return static_cast<const verilog_past_exprt &>(expr);
+}
+
+inline verilog_past_exprt &to_verilog_past_expr(exprt &expr)
+{
+  verilog_past_exprt::check(expr);
+  return static_cast<verilog_past_exprt &>(expr);
+}
```

Comment:
These should check that `expr.id() == ID_verilog_past`. (Are we perhaps missing this sort of check in some other place where this was initially copied from?)

Reply:
Fixed. Will look at the others.